ppl/zqd/postgres/errors.go

```diff
@@ -0,0 +1,15 @@
+package postgres
+
+import "github.com/go-pg/pg/v10"
+
+func ErrIsUniqueViolation(err error) bool {
+	// Would normally use the errors.As facilities but the does not work with
+	// certain queries in pkg pg.
+	pgerr, ok := err.(pg.Error)
+	return ok && pgerr.Field('C') == "23505"
+}
+
+func ErrIsForeignKeyViolation(err error) bool {
```

Comment:
Maybe call these IsUniqueViolation and IsForeignKeyViolation, like os.IsNotExist et al.?